src/scene/mesh.js

```diff
@@ -260,6 +279,19 @@ class Mesh extends RefCountedObject {
      * normally called by {@link Model#destroy} and does not need to be called manually.
      */
     destroy() {
+
+        const morph = this.morph;
+        if (morph) {
+
+            // this decreases ref count on the morph
+            this.morph = null;
+
+            // destroy morph
+            if (morph.getRefCount() < 1) {

```

Comment:
couldn't `RefCountedObject` do this automatically?

Reply:
That was my initial implementation a while back .. and for some forgotten reason it wasn't ideal so I removed it - we had some PR discussions around this as well I think.